qa/testing.go

```diff
@@ -344,6 +351,11 @@ func (f ResourceFixture) Apply(t *testing.T) (*schema.ResourceData, error) {
 	return resourceData, err
 }
 
+// ApplyCheckDiff not only applies the test case from fixture, but also takes in a diffChecker function and asserts the output of the diffChecker function to be true.
+func (f ResourceFixture) ApplyCheckDiff(t *testing.T, diffChceker func(*terraform.InstanceDiff) bool) (*schema.ResourceData, error) {
+	return f.applyInternal(t, diffChceker)
```

Comment:
Did you consider adding DiffChecker to ResourceFixture? New `Apply*` methods diverge the behavior between different cases. Suppose we want to do some other check after `Apply()`, like verify the final state. Would we want `ApplyCheckDiffAndState` and `ApplyCheckState` methods? Similarly, note that there aren't `ApplyRead/ApplyUpdate/ApplyCreate/ApplyDelete` methods. Almost all logic is in `Apply` itself. 

A better design would be to add this as a field to `ResourceFixture` itself. This is backwards compatible (no current test specifies this) and can be combined with current functionality easily.

Reply:
This makes sense to me, I can update this